libs/portal-integration-angular/src/lib/services/portal-dialog.service.ts

```diff
@@ -1,8 +1,424 @@
+ */
+export type DialogState<T> = {
+  button: 'primary' | 'secondary'
+  result: T | undefined
+}
+
+@Injectable({ providedIn: 'any' })
+export class PortalDialogService {
+  constructor(private dialogService: DialogService, private translateService: TranslateService) {}
+
+  /**
+   * @deprecated
+   */
+  open(componentType: Type<any>, config: DynamicDialogConfig): DynamicDialogRef {
+    if (isDevMode()) {
+      console.warn('You are using a deprecated method to display a dialog. Please move to the new one')
+    }
+    return this.dialogService.open(componentType, config)
+  }
+
+  /**
+   * Opens dialog with a component or message to display and one or two buttons. This method allows you to customize the dialog using parameters and by implementic specific interfaces via component to be displayed.
+   *
+   * @param title Dialog title
+   * @param componentOrMessage Dialog content to display
+   * @param primaryButtonTranslationKeyOrDetails Primary button to display in the bottom of the dialog
+   * @param secondaryButtonTranslationKeyOrDetails Secondary button to display in the bottom of the dialog
+   * @param showXButton Determines if X (close) button should be visible in the top right corner of the dialog (not displayed if set to false and displayed if set to true). NOTE: If only one button is being used then close button will not be visible on the dialog
+   * @returns Observable containing dialog state on close
+   *
+   * Displaying component inisde the dialog can be achieved by providing the component class with optional inputs. By default the component will be shown without any interaction with the dialog, however you can implement the following interfaces by your component class to allow for some interactions:
+   * - {@link DialogResult} - dialog state will contain dialogResult property
+   *
+   * - {@link DialogButtonClicked} - on button click ocxDialogButtonClicked function will be called with dialog state as a parameter. You should return true if you want dialog to be close or false if not and add any operations on your component.
+   *
+   * - {@link DialogPrimaryButtonDisabled} - dialog will use the EventEmitter to determine if the primary button should be disabled
+   *
+   * - {@link DialogSecondaryButtonDisabled} - dialog will use the EventEmitter to determine if the secondary button should be disabled
+   *
+   * @example
+   * Display dialog with message and two buttons using translation keys
+   *
+   * ```
+   * this.portalDialogService.openDialog('TITLE_KEY', 'WELCOME_MESSAGE', 'OK_BUTTON', 'REFRESH_BUTTON')
+   * ```
+   *
+   * @example
+   * Display dialog message with icon and single button
+   *
+   * ```
+   * // Welcome message with question mark icon
+   * const dialogMessage = {
+   *   key: 'WELCOME_MESSAGE',
+   *   icon: 'pi pi-question'
+   * }
+   * this.portalDialogService.openDialog('TITLE_KEY', dialogMessage, 'OK_BUTTON', 'REFRESH_BUTTON')
```

Comment:
Remove refresh button. This would be the second button therefore not matching the example